core/src/test/scala/org/apache/spark/sql/delta/optimize/OptimizeCompactionSuite.scala

```diff
@@ -437,6 +437,34 @@ trait OptimizeCompactionSuiteBase extends QueryTest
     }
   }
 
+  test("optimize command with multiple partition predicates with multiple where") {
+    withTempDir { tempDir =>
+      def writeData(count: Int): Unit = {
+        spark.range(count).select('id, lit("2017-10-10").cast("date") as 'date, 'id % 5 as 'part)
```

Comment:
Does this test really verify the change? Since we always use `2017-10-10` for the `date` column, adding `.where("date = '2017-10-10'")` will make no difference.